src/main/java/hudson/scm/SubversionSCM.java

```diff
@@ -1099,7 +1106,14 @@ public ISVNAuthenticationProvider createAuthenticationProvider(Job<?, ?> inConte
         return CredentialsSVNAuthenticationProviderImpl.createAuthenticationProvider(inContextOf, this, location);
     }
 
-
+    @Deprecated
+    public ISVNAuthenticationProvider createAuthenticationProvider(AbstractProject<?,?> inContextOf, ModuleLocation location) {
+        SubversionSCM scm = null;
+        if (inContextOf != null && inContextOf.getScm() instanceof SubversionSCM) {
+            scm = (SubversionSCM)inContextOf.getScm();
+        }
+        return CredentialsSVNAuthenticationProviderImpl.createAuthenticationProvider(inContextOf, scm, location);
```

Comment:
Passing null SCM may be risky, because the behavior is not documented. BTW, the implementations handle this null, so it's not very important